stan/math/rev/functor/operands_and_partials.hpp

```diff
@@ -371,17 +316,11 @@ class ops_partials_edge<double, std::vector<var_value<Op>>,
  private:
   template <typename, typename, typename, typename, typename, typename>
   friend class stan::math::operands_and_partials;
-  const std::vector<var_value<Op>>& operands_;
+  std::vector<var_value<Op>, arena_allocator<var_value<Op>>> operands_;
 
-  void dump_operands(vari** varis) {}
-  void dump_partials(double* partials) {}
-  int size() { return 0; }
-  std::tuple<const std::vector<var_value<Op>>&> container_operands() {
-    return std::forward_as_tuple(operands_);
-  }
-  std::tuple<partials_t&> container_partials() {
-    return std::forward_as_tuple(partials_vec_);
-  }
+  static constexpr int size() noexcept { return 0; }
+  inline auto&& operand() noexcept { return std::move(this->operands_); }
+  inline auto&& partial() noexcept { return std::move(this->partials_vec_); }
```

Comment:
So these are a trouble / nasty spot I need to think about more. What's going on here is that when I turned on clang sanitizer and had these defined as

```cpp
  inline auto operand() noexcept { return this->operands_; }
  inline auto partial() noexcept { return this->partials_vec_; }
```

or as 

```cpp
  inline auto& operand() noexcept { return this->operands_; }
  inline auto& partial() noexcept { return this->partials_vec_; }
```

I was getting memory leak errors. But, we know that after we call each of these functions we are not longer going to need the `std::vectors` that are in this class, so we just force a move to happen, thus when copying into the lambda we construct the operand and partial copy actually using the move constructor.

That seems odd, or at least seems odd we need to do that. I _think_ this might be a case just for std vectors of std vectors, but I need to look harder. I saw your post above about how by the standard it looks like we are doing everything correct, but this was just odd to me and after fighting mingws tuple implementation for so long I'm less trustworthy that the std impl is following all these things strictly. But it could also be on our end so I need to debug these a bit more to make sure we are doing the right thing here

Reply:
Update: I'm dumb and just needed to change

```cpp
  using partials_t = std::vector<Op, arena_allocator<Op>>;
```

to 

```cpp
  using partials_t = std::vector<arena_t<Op>, arena_allocator<arena_t<Op>>>;
```

and got rid of the move's etc.